src/Symfony/Bundle/SecurityBundle/Security/ContextAwareFirewallMapInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Security;
+
+use Symfony\Component\Security\Http\FirewallMapInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * This interface must be implemented by firewall maps that are able to return a
+ * context.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+interface ContextAwareFirewallMapInterface extends FirewallMapInterface
```

Comment:
@fabpot The interface has been removed. If it makes sense, I will adapt #19398 for typehint against `FirewallMap` (framework) directly instead of the checking the interface.